dataactvalidator/validation_handlers/validator.py

```diff
@@ -185,25 +192,61 @@ def cross_validate_sql(rules, submission_id, short_to_long_dict, first_file, sec
 
             # materialize as we'll iterate over the failed_rows twice
             failed_rows = list(failed_rows)
-            flex_data = relevant_cross_flex_data(failed_rows, submission_id, [first_file, second_file])
-
-            for row in failed_rows:
-                # get list of values for each column
-                values = ["{}: {}".format(short_to_long_dict[c], str(row[c])) if c in short_to_long_dict else
-                          "{}: {}".format(c, str(row[c])) for c in cols]
-                values = ", ".join(values)
-                full_column_string = column_string
-                # go through all flex fields in this row and add to the columns and values
-                for field in flex_data[row['row_number']]:
-                    full_column_string += ", " + field.header + "_file" +\
-                                          FILE_TYPE_DICT_LETTER[field.file_type_id].lower()
-                    values += ", {}: {}".format(field.header + "_file" +
-                                                FILE_TYPE_DICT_LETTER[field.file_type_id].lower(), field.cell)
-
-                target_file_type = FILE_TYPE_DICT_ID[rule.target_file_id]
-                failures.append([rule.file.name, target_file_type, full_column_string,
-                                str(rule.rule_error_message), values, row['row_number'], str(rule.rule_label),
-                                rule.file_id, rule.target_file_id, rule.rule_severity_id])
+            num_failed_rows = len(failed_rows)
+            slice_start = 0
+            slice_size = 10000
+            while slice_start <= num_failed_rows:
+                failed_row_subset = failed_rows[slice_start:slice_start+slice_size]
+                # finding out row numbers for logger
+                last_error_curr_slice = slice_start + slice_size
```

Comment:
Might as well move this up one and use it in the failed_row_subset calculation, to only do the addition once